matrix_client/client.py

```diff
@@ -222,36 +221,44 @@ def login_with_password_no_sync(self, username, password):
         Raises:
             MatrixRequestError
         """
-        response = self.api.login(
-            "m.login.password", user=username, password=password
-        )
-        self.user_id = response["user_id"]
-        self.token = response["access_token"]
-        self.hs = response["home_server"]
-        self.api.token = self.token
-        return self.token
-
-    def login_with_password(self, username, password, limit=10):
+        return self.login_with_password(username, password, sync=False)
+
+    def login_with_password(self,
+                            username,
+                            password,
+                            limit=10,
+                            sync=True,
+                            device_id=None):
         """ Login to the homeserver.
 
         Args:
             username (str): Account username
             password (str): Account password
             limit (int): Deprecated. How many messages to return when syncing.
                 This will be replaced by a filter API in a later release.
+            sync (bool): Optional. Whether to initiate a /sync request after logging in.
+            device_id (str): Optional. ID of the client device. The server will
+                auto-generate a device_id if this is not specified.
 
         Returns:
             str: Access token
 
         Raises:
             MatrixRequestError
         """
-        token = self.login_with_password_no_sync(username, password)
```

Comment:
Let's stick a deprecation warning here, too, and move all this functionality to a vanilla `login` method. Eventually both `username` and `password` will be kwargs and an additional kwarg `token` will be introduced to allow `m.login.token` with this same method.